catalog/service/common/src/main/java/org/projectnessie/catalog/service/config/CatalogConfig.java

```diff
@@ -57,11 +73,21 @@ public interface CatalogConfig {
    */
   default WarehouseConfig getWarehouse(String warehouse) {
     boolean hasWarehouse = warehouse != null && !warehouse.isEmpty();
-    if (hasWarehouse) {
-      WarehouseConfig w = warehouses().get(warehouse);
-      if (w != null) {
-        return w;
+    if (!hasWarehouse) {
+      Optional<String> def = defaultWarehouse();
```

Comment:
why not call `resolveWarehouseName(warehouse)`?

Reply:
done